storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -299,6 +310,68 @@ func (d *DotGit) addRefsFromPackedRefs(refs *[]*plumbing.Reference) (err error)
 	return s.Err()
 }
 
+func (d *DotGit) rewritePackedRefsWithoutRef(name plumbing.ReferenceName) (err error) {
+	f, err := d.fs.Open(packedRefsPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+
+		return err
+	}
+	defer ioutil.CheckClose(f, &err)
+
+	tmp, err := d.fs.TempFile("tmp", "packedrefs")
+	if err != nil {
+		return err
+	}
+	defer ioutil.CheckClose(tmp, &err)
+	defer os.Remove(tmp.Filename())
+
+	s := bufio.NewScanner(f)
+	found := false
+	for s.Scan() {
+		line := s.Text()
+		ref, err := d.processLine(line)
+		if err != nil {
+			return err
+		}
+
+		if ref != nil && ref.Name() == name {
+			found = true
+			continue
+		}
+
+		if _, err := fmt.Fprintln(tmp, line); err != nil {
+			return err
+		}
+	}
+
+	if !found {
+		return nil
+	}
+
+	if err := s.Err(); err != nil {
```

Comment:
I think this check must go before 350, so in case of error and not found we actually return the error instead of nil.